cctp_primitives/src/bit_vector/compression.rs

```diff
@@ -186,10 +210,19 @@ fn gzip_compress(bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
     Ok(result)
 }
 
-fn gzip_decompress(compressed_bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
-    let mut uncompressed_bitvector = Vec::new();
-    let mut e = GzDecoder::new(compressed_bit_vector);
-    e.read_to_end(&mut uncompressed_bitvector)?;
+fn gzip_decompress(compressed_bit_vector: &[u8], max_decompressed_size: usize) -> Result<Vec<u8>, Error> {
+    let mut uncompressed_bitvector = Vec::with_capacity(max_decompressed_size);
+    let mut decompressor = GzDecoder::new(compressed_bit_vector);
+    let mut fixed_array = [0; DECOMPRESSION_CHUNK_SIZE];
+
+    loop {
```

Comment:
Maybe comment the rationale behind this piece of code

Reply:
Fixed.